src/bytes.rs

```diff
@@ -338,13 +439,26 @@ impl BytesVt {
     }
 }
 
-#[cfg(feature = "alloc")]
+#[cfg(all(feature = "alloc", not(feature = "stabby")))]
+/// Clones `self` by either:
+/// - Performing a reference count increment (or a no-op for references) if possible
+/// - Copying the data in a new `Arc<[u8]>` and wrapping it in a new [`Bytes`]
 impl Clone for Bytes<'_> {
     fn clone(&self) -> Self {
         self.noalloc_clone()
             .unwrap_or_else(|| Arc::<[u8]>::from(self.as_slice()).into())
     }
 }
+#[cfg(feature = "stabby")]
+/// Clones `self` by either:
+/// - Performing a reference count increment (or a no-op for references) if possible
+/// - Copying the data in a new [`stabby::sync::ArcSlice<u8>`](stabby::sync::ArcSlice) and wrapping it in a new [`Bytes`]
+impl Clone for Bytes<'_> {
+    fn clone(&self) -> Self {
+        self.noalloc_clone()
+            .unwrap_or_else(|| stabby::sync::ArcSlice::<u8>::from(self.as_slice()).into())
+    }
+}
```

Comment:
If we move my `match_cfg!` / `cfg`-generic `ArcSlice` definition we could avoid the duplication of doc and signature here as well. However, I think that something even better would be to move that `&[u8] -> Bytes<'static> { <refcounted> }` logic to its own API:

```rs
impl Bytes<'static> {
    #[cfg(any(feature = "alloc", feature = "stabby"))]
    /// Docs... "some heap-allocated and ref-counted slice type", either `Arc` or `ArcSlice` etc.
    pub fn upgraded_from_slice(slice: &[u8]) -> Bytes<'static> {
        match_cfg! {
            …
        }
        ArcSlice::<u8>::from(slice).into()
    }
}
```

so as to use this in `fn upgrade(self: Bytes<'a>) -> Bytes<'static> {`, as well as here:

```rs
#[cfg(any(feature = "alloc", feature = "stabby"))]
/// Clones `self` by either:
/// - Performing a reference count increment (or a no-op for references) if possible
/// - Copying the data in a new `Arc<[u8]>` and wrapping it in a new [`Bytes`]
impl Clone for Bytes<'_> {
    fn clone(&self) -> Self {
        self.noalloc_clone()
            .unwrap_or_else(|| Bytes::upgraded_from_slice(self.as_slice()))
    }
}
```